File: cpp-tiny/lib/Models/MissingArgumentErrorAllOf.cpp


#include "MissingArgumentError_allOf.h"

using namespace Tiny;

MissingArgumentError_allOf::MissingArgumentError_allOf()
{
	result = null;
	msg = null;
	code = null;
	var_name = std::string();
}

MissingArgumentError_allOf::MissingArgumentError_allOf(std::string jsonString)
{
	this->fromJson(jsonString);
}

MissingArgumentError_allOf::~MissingArgumentError_allOf()
{

}

void
MissingArgumentError_allOf::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *resultKey = "result";
    
    if(object.has_key(resultKey)) 
    {
        bourne::json value = object[resultKey];


        

        AnyType* obj = &result;
		obj->fromJson(value.dump());

    }

    const char *msgKey = "msg";
    
    if(object.has_key(msgKey)) 
    {
        bourne::json value = object[msgKey];


        

        AnyType* obj = &msg;
		obj->fromJson(value.dump());

    }

    const char *codeKey = "code";
    
    if(object.has_key(codeKey)) 
    {
        bourne::json value = object[codeKey];


        

        AnyType* obj = &code;
		obj->fromJson(value.dump());

    }

    const char *var_nameKey = "var_name";
    
    if(object.has_key(var_nameKey)) 
    {
        bourne::json value = object[var_nameKey];


        
        jsonToValue(&var_name, value, "std::string");


    }


}

bourne::json
MissingArgumentError_allOf::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["result"] = getResult().toJson();

    




	object["msg"] = getMsg().toJson();

    




	object["code"] = getCode().toJson();

    



    object["var_name"] = getVarName();



    return object;

}

AnyType
MissingArgumentError_allOf::getResult()
{
	return result;
}

void
MissingArgumentError_allOf::setResult(AnyType  result)
{
	this->result = result;
}

AnyType
MissingArgumentError_allOf::getMsg()
{
	return msg;
}

void
MissingArgumentError_allOf::setMsg(AnyType  msg)
{
	this->msg = msg;
}

AnyType
MissingArgumentError_allOf::getCode()
{
	return code;
}

void
MissingArgumentError_allOf::setCode(AnyType  code)
{
	this->code = code;
}

std::string
MissingArgumentError_allOf::getVarName()
{
	return var_name;
}

void
MissingArgumentError_allOf::setVarName(std::string  var_name)
{
	this->var_name = var_name;
}

File: java-play-framework/app/apimodels/MissingArgumentErrorAllOf.java
package apimodels;

import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import java.util.Objects;
import javax.validation.constraints.*;
/**
 * ## Missing request parameter(s)  A typical failed JSON response for when a required request parameter is not supplied 
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen")
@SuppressWarnings({"UnusedReturnValue", "WeakerAccess"})
public class MissingArgumentErrorAllOf   {
  @JsonProperty("result")
  
  private Object result = null;

  @JsonProperty("msg")
  
  private Object msg = null;

  @JsonProperty("code")
  
  private Object code = null;

  @JsonProperty("var_name")
  
  private String varName;

  public MissingArgumentErrorAllOf result(Object result) {
    this.result = result;
    return this;
  }

   /**
   * Get result
   * @return result
  **/
  public Object getResult() {
    return result;
  }

  public void setResult(Object result) {
    this.result = result;
  }

  public MissingArgumentErrorAllOf msg(Object msg) {
    this.msg = msg;
    return this;
  }

   /**
   * Get msg
   * @return msg
  **/
  public Object getMsg() {
    return msg;
  }

  public void setMsg(Object msg) {
    this.msg = msg;
  }

  public MissingArgumentErrorAllOf code(Object code) {
    this.code = code;
    return this;
  }

   /**
   * Get code
   * @return code
  **/
  public Object getCode() {
    return code;
  }

  public void setCode(Object code) {
    this.code = code;
  }

  public MissingArgumentErrorAllOf varName(String varName) {
    this.varName = varName;
    return this;
  }

   /**
   * It contains the information about the missing parameter. 
   * @return varName
  **/
  public String getVarName() {
    return varName;
  }

  public void setVarName(String varName) {
    this.varName = varName;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MissingArgumentErrorAllOf missingArgumentErrorAllOf = (MissingArgumentErrorAllOf) o;
    return Objects.equals(result, missingArgumentErrorAllOf.result) &&
        Objects.equals(msg, missingArgumentErrorAllOf.msg) &&
        Objects.equals(code, missingArgumentErrorAllOf.code) &&
        Objects.equals(varName, missingArgumentErrorAllOf.varName);
  }

  @Override
  public int hashCode() {
    return Objects.hash(result, msg, code, varName);
  }

  @SuppressWarnings("StringBufferReplaceableByString")
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class MissingArgumentErrorAllOf {\n");
    
    sb.append("    result: ").append(toIndentedString(result)).append("\n");
    sb.append("    msg: ").append(toIndentedString(msg)).append("\n");
    sb.append("    code: ").append(toIndentedString(code)).append("\n");
    sb.append("    varName: ").append(toIndentedString(varName)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

